src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -16580,7 +16580,17 @@ def index_equals(self, other: "SnowflakeQueryCompiler") -> bool:
             )
         }
 
-        rows = join_result.result_frame.ordered_dataframe.agg(agg_exprs).collect()
+        try:
+            rows = join_result.result_frame.ordered_dataframe.agg(agg_exprs).collect()
+        except SnowparkSQLException as e:
```

Comment:
I'm not sure if there are different error messages for different type casts. I've seen something like:
```
SnowparkSQLException: (1304): 01b1d9a9-0507-c7de-000c-a9033d54baf7: 100037 (22018): Boolean value 'bad value' is not recognized
```
before (this is from Rehan's fillna type casting doc).
I'm not sure if this error can be raised and if there are similar type errors but with different text output. Thought it was worth pointing out.

Reply:
good point, I think it's better to loose the condition here then.